sled-hardware/src/disk.rs

```diff
@@ -132,6 +132,46 @@ impl DiskPaths {
     }
 }
 
+// XXX MTZ: Make this an enum with DiskFirmware::Nvme?
+#[derive(
+    Debug, Clone, PartialEq, Eq, Hash, Ord, PartialOrd, Deserialize, Serialize,
+)]
+pub struct DiskFirmware {
+    active_slot: u8,
+    next_active_slot: Option<u8>,
+    slot1_read_only: bool,
+    slots: Vec<Option<String>>,
```

Comment:
What's the value of this String? This feels a little bit like a more specific type that has been coerced into a string?

Reply:
See the above comment.